media-gfx/nomacs/files/nomacs-3.17.2282-exiv-0.28.0-build-fix.patch

```diff
@@ -0,0 +1,274 @@
+         pos = exifData.findKey(key);
+     }
+ 
++#if EXIV2_TEST_VERSION(0, 28, 0)
++    Exiv2::Value::UniquePtr v = pos->getValue();
++#else
+     Exiv2::Value::AutoPtr v = pos->getValue();
++#endif
+     Exiv2::UShortValue *prv = dynamic_cast<Exiv2::UShortValue *>(v.release());
+     if (!prv)
+         return;
+ 
++#if EXIV2_TEST_VERSION(0, 28, 0)
++    Exiv2::UShortValue::UniquePtr rv = Exiv2::UShortValue::UniquePtr(prv);
++#else
+     Exiv2::UShortValue::AutoPtr rv = Exiv2::UShortValue::AutoPtr(prv);
++#endif
+     if (rv->value_.empty())
+         return;
+ 
+@@ -1202,7 +1267,11 @@ void DkMetaDataT::setRating(int r)
+         exifData["Exif.Image.Rating"] = uint16_t(r);
+         exifData["Exif.Image.RatingPercent"] = uint16_t(r);
+ 
++#if EXIV2_TEST_VERSION(0, 28, 0)
++        Exiv2::Value::UniquePtr v = Exiv2::Value::create(Exiv2::xmpText);
++#else
+         Exiv2::Value::AutoPtr v = Exiv2::Value::create(Exiv2::xmpText);
++#endif
+         v->read(sRating);
+         xmpData.add(Exiv2::XmpKey("Xmp.xmp.Rating"), v.get());
+         v->read(sRatingPercent);
+@@ -1448,9 +1517,17 @@ DkRotatingRect DkMetaDataT::getXMPRect(const QSize &size) const
+     return DkRotatingRect(rr);
+ }
+ 
++#if EXIV2_TEST_VERSION(0, 28, 0)
++Exiv2::Image::UniquePtr DkMetaDataT::loadSidecar(const QString &filePath) const
++#else
+ Exiv2::Image::AutoPtr DkMetaDataT::loadSidecar(const QString &filePath) const
++#endif
+ {
++#if EXIV2_TEST_VERSION(0, 28, 0)
++    Exiv2::Image::UniquePtr xmpImg;
++#else
+     Exiv2::Image::AutoPtr xmpImg;
++#endif
+ 
+     // TODO: check if the file type supports xmp
+ 
+@@ -1500,7 +1577,11 @@ bool DkMetaDataT::setXMPValue(Exiv2::XmpData &xmpData, QString xmpKey, QString x
+         if (!pos->setValue(xmpValue.toStdString()))
+             setXMPValueSuccessful = true;
+     } else {
++#if EXIV2_TEST_VERSION(0, 28, 0)
++        Exiv2::Value::UniquePtr v = Exiv2::Value::create(Exiv2::xmpText);
++#else
+         Exiv2::Value::AutoPtr v = Exiv2::Value::create(Exiv2::xmpText);
++#endif
+         if (!v->read(xmpValue.toStdString())) {
+             if (!xmpData.add(Exiv2::XmpKey(key), v.get()))
+                 setXMPValueSuccessful = true;
+--- a/src/DkCore/DkMetaData.h
++++ b/src/DkCore/DkMetaData.h
+@@ -31,6 +31,7 @@
+ #include <QMap>
+ #include <QSharedPointer>
+ #include <QStringList>
++#include <exiv2/exiv2.hpp>
+ 
+ // code for metadata crop:
+ #include "DkMath.h"
+@@ -155,7 +156,11 @@ public:
+     bool setXMPValue(Exiv2::XmpData &xmpData, QString xmpKey, QString xmpValue);
+ 
+ protected:
++#if EXIV2_TEST_VERSION(0, 28, 0)
++    Exiv2::Image::UniquePtr loadSidecar(const QString &filePath) const;
++#else
+     Exiv2::Image::AutoPtr loadSidecar(const QString &filePath) const;
++#endif
+ 
+     enum {
+         not_loaded,
+@@ -164,7 +169,11 @@ protected:
+         dirty,
+     };
+ 
++#if EXIV2_TEST_VERSION(0, 28, 0)
++    Exiv2::Image::UniquePtr mExifImg; // TODO std::unique_ptr<Exiv2::Image> (and all other *::AutoPtr)
```

Comment:
`// TODO std::unique_ptr<Exiv2::Image> (and all other *::AutoPtr)`

See also:
https://github.com/openstreetmap/merkaartor/commit/1e20d2ccd743ea5f8c2358e4ae36fead8b9390fd

Using that would avoid `#ifdef`s and maybe result in an upstream-presentable PR for them to consider?

Alternatively, `use auto`: https://invent.kde.org/graphics/kphotoalbum/-/commit/82520a00bb8ca3f6400c7cc053066072e5055594